### PR TITLE
Add transport helper functionality

### PR DESCRIPTION
**Pull request by dkmyta**
#### Description
- Currently, Jetpack and the standalone Jetpack Backup plugin rely on the `backup` package for helper script installation, deletion, and clearing. With the introduction of Jetpack Protect and a third plugin that will require this functionality, we have decided to separate out and generalize all helper script related code (endpoints, cron, and other management) into a separate `transport-helper` package so that the tooling can be easily shared between the three reliant plugins and any others that may have the requirement in the future.
- The intention of this initial pass is to create a package that only contains the helper script specific code from the `backup` package, that will work in parallel until a time when we are confident that the code can be removed for the `backup` package and all reliant internal/external code has been updated to reflect the change so that we can confidently rely solely on the `transport-helper` package for these actions.

#### Changes proposed in this Pull Request:

- [x] Create a new `transport-helper` package (#25772)
- [x] Duplicate and generalize all helper script code to the `transport-helper` package, with the exception of the registered endpoint route `/backup-helper-script`
- [x] Update WPcom code reliant on the `backup` package for processing helper script requests to account for the new package (D87446-code)

#### Next Steps:
- Remove any/all helper script code from the `backup` package, and generalize the registered endpoint route to `/helper-script`
- Update any/all internal code reliant on the helper script code from the `backup` package accordingly
- Update any/all external code reliant on the helper script code from the `backup` package accordingly

#### Other information:
- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

- 1201069996155224-as-1202645981599451
- p1659461330589659-slack-CS8UYNPEE

#### Does this pull request change what data or activity we track or use?

- No

#### Testing instructions:

* With 87446-code now deployed, proceed with the following:
* Start up a Jurassic Ninja instance, only including the Jetpack Beta plugin and this Protect branch
* Add a Scan plan to the site
* Open up the RWDB and verify that the `auto-http` protocol cxn tests pass
* Queue a scan and verify that it succeeds

### CODE REVIEW

File: projects/packages/transport-helper/actions.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+		);
+	};
+	$add_action = $add_filter;
+}
+
+// Clean up expired Jetpack Helper Scripts from a scheduled event.
+$add_action( 'jetpack_cleanup_helper_scripts', array( 'Automattic\\Jetpack\\Transport_Helper\\Helper_Script_Manager', 'cleanup_expired_helper_scripts' ) );
+
+// Register REST routes.
+$add_action( 'rest_api_init', array( 'Automattic\\Jetpack\\Transport_Helper\\\REST_Controller', 'register_rest_routes' ) );
```
Comment (miguelxpn) [inline projects/packages/transport-helper/actions.php]: There's an extra \ before the REST_Controller

File: projects/packages/transport-helper/actions.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+// Set up package version hook.
+$add_filter( 'jetpack_package_versions', 'Automattic\\Jetpack\\Transport_Helper\\\Package_Version::send_package_version_to_tracker' );
```
Comment (miguelxpn) [inline projects/packages/transport-helper/actions.php]: Another extra \ here

File: projects/packages/transport-helper/composer.json [hunk 3]
```diff
@@ -3,7 +3,9 @@
 	"description": "Package to help transport server communication",
```
Comment (miguelxpn) [inline projects/packages/transport-helper/composer.json:3]: We need to add the actions.php file to the composer autoload so we run the file when we require the package, like the backups one.

**Review by miguelxpn — DISMISSED** [review]

File: projects/packages/transport-helper/src/class-helper-script-manager.php [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+
+namespace Automattic\Jetpack\Transport_Helper;
+
+/**
+ * Helper_Script_Manager manages installation, deletion and cleanup of Helper Scripts
+ */
+class Helper_Script_Manager {
+
+	const TEMP_DIRECTORY = 'jetpack-temp';
+	const HELPER_HEADER  = "<?php /* Jetpack Helper Script */\n";
```
Comment (miguelxpn) [inline projects/packages/transport-helper/src/class-helper-script-manager.php]: This header should be `<?php /* Jetpack Backup Helper Script */\n`

**Review by miguelxpn — APPROVED** [review]
LGTM!

**Review by bisko — APPROVED** [review]
LGTM, tests as per the testing instructions.

**Pull request merged**